crates/vsock/src/vsock_conn.rs

```diff
@@ -308,8 +308,13 @@ impl<S: AsRawFd + Read + Write> VsockConnection<S> {
         if written_count > 0 {
             // Increment forwarded count by number of bytes written to the stream
             self.fwd_cnt += Wrapping(written_count as u32);
-            // TODO: https://github.com/torvalds/linux/commit/c69e6eafff5f725bc29dcb8b52b6782dca8ea8a2
-            self.rx_queue.enqueue(RxOps::CreditUpdate);
+
+            // We send a credit update only when the space available seen
+            // by the transmitter is less than CREDIT_UPDATE_THRESHOLD.
+            let free_space = Wrapping(CONN_TX_BUF_SIZE) - (self.fwd_cnt - self.last_fwd_cnt);
+            if free_space.0 < CREDIT_UPDATE_THRESHOLD {
+                self.rx_queue.enqueue(RxOps::CreditUpdate);
+            }
```

Comment:
Personal opinion, I find `wrapping_sub` to be more readable and simple, than [`Wrapping`](https://doc.rust-lang.org/std/num/struct.Wrapping.html).
```suggestion
            let free_space = CONN_TX_BUF_SIZE.wrapping_sub(self.fwd_cnt - self.last_fwd_cnt);
            if free_space < CREDIT_UPDATE_THRESHOLD {
                self.rx_queue.enqueue(RxOps::CreditUpdate);
            }
```